protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java

```diff
@@ -409,7 +337,7 @@ public synchronized void event(Event event) {
                     final List<MessageUid> uids = ((Added) event).getUids();
                     SelectedMailbox sm = session.getSelected();
                     for (MessageUid uid : uids) {
-                        add(uid);
+                        uidMsnMapper.get().addUid(uid);
```

Comment:
Hmm I disagree with initializing upon events. We (I belive) don't need to update until you need it...